pkg/reporter/sarif_reporter.go

```diff
@@ -0,0 +1,96 @@
+package reporter
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/google/osv-scanner/internal/output"
+	"github.com/google/osv-scanner/pkg/models"
+	"github.com/owenrumney/go-sarif/v2/sarif"
```

Comment:
Are there any alternatives to using this dependency? Is this a significant amount of code to be writing ourselves? 

e.g. Scorecard seems to have implemented their own sarif support: https://github.com/ossf/scorecard/blob/eb941e396499e64e7b83d30e54c0a462c93f61fb/pkg/sarif.go#L155

Reply:
The scorecard implementation is 700+ lines and 800+ lines of tests. We can try vendoring that implementation in?

Reply:
Discussed offline, seems like the transitive dependencies are relatively minimal so let's just stick with using this library.